clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -53,6 +50,15 @@
         }
     };
 
+    // mutable state
```

Comment:
They are final, but mutable. The idea was to highlight the mutable fields that this class maintains. The rest are either truly immutable or just views. If this is more confusing than helpful, I can remove it.